tests/components/control4/test_climate.py

```diff
@@ -0,0 +1,502 @@
+            "HEAT_SETPOINT_F": 68.0,
+        }
+    }
+
+    async def mock_update_vars(*args, **kwargs):
+        return climate_vars
+
+    with (
+        patch(
+            "homeassistant.components.control4.climate.update_variables_for_config_entry",
+            new=mock_update_vars,
+        ),
+        patch("homeassistant.components.control4.PLATFORMS", platforms),
+    ):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    with (
+        patch(
+            "pyControl4.climate.C4Climate.setHeatSetpointF", new_callable=AsyncMock
+        ) as mock_heat,
+        patch(
+            "pyControl4.climate.C4Climate.setCoolSetpointF", new_callable=AsyncMock
+        ) as mock_cool,
+    ):
+        await hass.services.async_call(
+            CLIMATE_DOMAIN,
+            SERVICE_SET_TEMPERATURE,
+            {
+                ATTR_ENTITY_ID: ENTITY_ID,
+                "target_temp_low": 65.0,
+                "target_temp_high": 78.0,
+            },
+            blocking=True,
+        )
+        mock_heat.assert_called_once_with(65.0)
+        mock_cool.assert_called_once_with(78.0)
+
+
+@pytest.mark.usefixtures("mock_c4_account", "mock_c4_director")
+async def test_climate_not_created_when_no_initial_data(
+    hass: HomeAssistant,
+    mock_config_entry,
+    platforms,
+) -> None:
+    """Test climate entity is not created when coordinator has no initial data."""
+    hass.config.units = US_CUSTOMARY_SYSTEM
+    mock_config_entry.add_to_hass(hass)
+
+    async def mock_update_vars(*args, **kwargs):
+        return {}
+
+    with (
+        patch(
+            "homeassistant.components.control4.climate.update_variables_for_config_entry",
+            new=mock_update_vars,
+        ),
+        patch("homeassistant.components.control4.PLATFORMS", platforms),
+    ):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    # Entity should not be created if there's no data during initial setup
+    state = hass.states.get(ENTITY_ID)
+    assert state is None
+
+
+@pytest.mark.usefixtures("mock_c4_account", "mock_c4_director")
+async def test_climate_missing_variables(
+    hass: HomeAssistant,
+    mock_config_entry,
+    platforms,
+) -> None:
+    """Test climate entity handles missing variables gracefully."""
+    hass.config.units = US_CUSTOMARY_SYSTEM
+    mock_config_entry.add_to_hass(hass)
+
+    climate_vars = {
+        123: {
+            "HVAC_STATE": "idle",
+            "HVAC_MODE": "Heat",
+            # Missing TEMPERATURE_F and HUMIDITY
+            "COOL_SETPOINT_F": 75.0,
+            "HEAT_SETPOINT_F": 68.0,
+        }
+    }
+
+    async def mock_update_vars(*args, **kwargs):
+        return climate_vars
+
+    with (
+        patch(
+            "homeassistant.components.control4.climate.update_variables_for_config_entry",
+            new=mock_update_vars,
+        ),
```

Comment:
Instead of repatching this, you can do
```py
@pytest.mark.parametrize("mock_climate_variables", [{
        123: {
            "HVAC_STATE": "idle",
            "HVAC_MODE": "Heat",
            # Missing TEMPERATURE_F and HUMIDITY
            "COOL_SETPOINT_F": 75.0,
            "HEAT_SETPOINT_F": 68.0,
        }
    }
])
```
and it will work :)